slick/src/main/scala/slick/lifted/TupleInSetImplicits.fm

```diff
@@ -0,0 +1,67 @@
+package slick.lifted
+
+import slick.ast._
+import ScalaBaseType._
+import slick.util.ConstArray
+import slick.lifted.FunctionSymbolExtensionMethods._
+import slick.lifted._
+import scala.language.{higherKinds, implicitConversions}
+import scala.reflect.ClassTag
+import slick.ast.{BaseTypedType, Library, LiteralNode, ProductNode, ScalaBaseType}
+import ScalaBaseType._
+import scala.language.{higherKinds, implicitConversions}
+import slick.util.ConstArray
+
+
+
+sealed trait TypeContext[B1, P1] {
```

Comment:
It seems that `TypeContext` has only one subtype `TupleTypeContext` which has many subtypes.
I guess that means the trait `TypeContext`  is not needed (as it can be replaced by `TupleTypeContext`)

Reply:
yes, TupeTypeContext seems more reasonable here